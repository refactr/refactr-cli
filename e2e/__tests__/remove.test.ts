import { executeWithStdoutOnly } from '../helpers/execute';
import { loadFixtures } from './fixtures';

beforeAll(async () => {
  return await loadFixtures();
});

describe('refactrctl remove', () => {
  test('throws on missing subcommand', async () => {
    await expect(executeWithStdoutOnly(['remove'])).rejects.toMatchSnapshot();
  });

  describe('credential', () => {
    test('throws on missing arguments', async () => {
      await expect(
        executeWithStdoutOnly(['remove', 'credential'])
      ).rejects.toMatchSnapshot();
    });

    test('throws on missing --project-id option', async () => {
      await expect(
        executeWithStdoutOnly(['remove', 'credential', 'credential-id'])
      ).rejects.toMatchSnapshot();
    });
  });

  describe('job', () => {
    test('throws on missing arguments', async () => {
      await expect(
        executeWithStdoutOnly(['remove', 'job'])
      ).rejects.toMatchSnapshot();
    });

    test('throws on missing --project-id option', async () => {
      await expect(
        executeWithStdoutOnly(['remove', 'job', 'job-id'])
      ).rejects.toMatchSnapshot();
    });
  });

  describe('pipeline', () => {
    test('throws on missing arguments', async () => {
      await expect(
        executeWithStdoutOnly(['remove', 'pipeline'])
      ).rejects.toMatchSnapshot();
    });
  });

  describe('project', () => {
    test('throws on missing arguments', async () => {
      await expect(
        executeWithStdoutOnly(['remove', 'project'])
      ).rejects.toMatchSnapshot();
    });
  });

  describe('runner', () => {
    test('throws on missing arguments', async () => {
      await expect(
        executeWithStdoutOnly(['remove', 'runner'])
      ).rejects.toMatchSnapshot();
    });

    test('throws on missing organization-id option', async () => {
      await expect(
        executeWithStdoutOnly(['remove', 'runner', 'runner-id'])
      ).rejects.toMatchSnapshot();
    });
  });
});
