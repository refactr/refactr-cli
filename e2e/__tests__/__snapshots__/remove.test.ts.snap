// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`refactrctl remove credential throws on missing --project-id option 1`] = `
"refactrctl remove credential <credential-id>

Delete specified runner

Positionals:
  credential-id  [string] [required]

Options:
      --version     Show version number  [boolean]
  -v, --verbose     Print detailed output  [boolean]
      --format      Specifies output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --address     Address of the Refactr server. This can also be specified via the REFACTR_ADDRESS environment variable  [string] [required] [default: \\"https://api.refactr.it/v1\\"]
      --auth-token  Authentication token. This can also be specified via the REFACTR_AUTH_TOKEN environment variable  [string]
      --help        Show help  [boolean]
      --project-id  Project this credential belongs to  [string] [required]

Missing required argument: project-id
"
`;

exports[`refactrctl remove credential throws on missing arguments 1`] = `
"refactrctl remove credential <credential-id>

Delete specified runner

Positionals:
  credential-id  [string] [required]

Options:
      --version     Show version number  [boolean]
  -v, --verbose     Print detailed output  [boolean]
      --format      Specifies output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --address     Address of the Refactr server. This can also be specified via the REFACTR_ADDRESS environment variable  [string] [required] [default: \\"https://api.refactr.it/v1\\"]
      --auth-token  Authentication token. This can also be specified via the REFACTR_AUTH_TOKEN environment variable  [string]
      --help        Show help  [boolean]
      --project-id  Project this credential belongs to  [string] [required]

Not enough non-option arguments: got 0, need at least 1
"
`;

exports[`refactrctl remove job throws on missing --project-id option 1`] = `
"refactrctl remove job <job-id>

Delete specified job

Positionals:
  job  [string]

Options:
      --version     Show version number  [boolean]
  -v, --verbose     Print detailed output  [boolean]
      --format      Specifies output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --address     Address of the Refactr server. This can also be specified via the REFACTR_ADDRESS environment variable  [string] [required] [default: \\"https://api.refactr.it/v1\\"]
      --auth-token  Authentication token. This can also be specified via the REFACTR_AUTH_TOKEN environment variable  [string]
      --help        Show help  [boolean]
      --project-id  Project this pipeline belongs to  [string] [required]

Missing required argument: project-id
"
`;

exports[`refactrctl remove job throws on missing arguments 1`] = `
"refactrctl remove job <job-id>

Delete specified job

Positionals:
  job  [string]

Options:
      --version     Show version number  [boolean]
  -v, --verbose     Print detailed output  [boolean]
      --format      Specifies output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --address     Address of the Refactr server. This can also be specified via the REFACTR_ADDRESS environment variable  [string] [required] [default: \\"https://api.refactr.it/v1\\"]
      --auth-token  Authentication token. This can also be specified via the REFACTR_AUTH_TOKEN environment variable  [string]
      --help        Show help  [boolean]
      --project-id  Project this pipeline belongs to  [string] [required]

Not enough non-option arguments: got 0, need at least 1
"
`;

exports[`refactrctl remove pipeline throws on missing arguments 1`] = `
"refactrctl remove pipeline <pipeline-id>

Delete specified pipeline

Positionals:
  pipeline-id  Pipeline to delete  [string] [required]

Options:
      --version     Show version number  [boolean]
  -v, --verbose     Print detailed output  [boolean]
      --format      Specifies output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --address     Address of the Refactr server. This can also be specified via the REFACTR_ADDRESS environment variable  [string] [required] [default: \\"https://api.refactr.it/v1\\"]
      --auth-token  Authentication token. This can also be specified via the REFACTR_AUTH_TOKEN environment variable  [string]
      --help        Show help  [boolean]
      --project-id  Project this pipeline belongs to  [string] [required]

Not enough non-option arguments: got 0, need at least 1
"
`;

exports[`refactrctl remove project throws on missing arguments 1`] = `
"refactrctl remove project <project-id>

Delete specified project

Positionals:
  project-id  [string] [required]

Options:
      --version     Show version number  [boolean]
  -v, --verbose     Print detailed output  [boolean]
      --format      Specifies output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --address     Address of the Refactr server. This can also be specified via the REFACTR_ADDRESS environment variable  [string] [required] [default: \\"https://api.refactr.it/v1\\"]
      --auth-token  Authentication token. This can also be specified via the REFACTR_AUTH_TOKEN environment variable  [string]
      --help        Show help  [boolean]

Not enough non-option arguments: got 0, need at least 1
"
`;

exports[`refactrctl remove runner throws on missing arguments 1`] = `
"refactrctl remove runner <runner-id>

Delete specified runner

Positionals:
  runner-id  [string] [required]

Options:
      --version          Show version number  [boolean]
  -v, --verbose          Print detailed output  [boolean]
      --format           Specifies output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --address          Address of the Refactr server. This can also be specified via the REFACTR_ADDRESS environment variable  [string] [required] [default: \\"https://api.refactr.it/v1\\"]
      --auth-token       Authentication token. This can also be specified via the REFACTR_AUTH_TOKEN environment variable  [string]
      --help             Show help  [boolean]
      --organization-id  Organization this runner belongs to  [string] [required]

Not enough non-option arguments: got 0, need at least 1
"
`;

exports[`refactrctl remove runner throws on missing organization-id option 1`] = `
"refactrctl remove runner <runner-id>

Delete specified runner

Positionals:
  runner-id  [string] [required]

Options:
      --version          Show version number  [boolean]
  -v, --verbose          Print detailed output  [boolean]
      --format           Specifies output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --address          Address of the Refactr server. This can also be specified via the REFACTR_ADDRESS environment variable  [string] [required] [default: \\"https://api.refactr.it/v1\\"]
      --auth-token       Authentication token. This can also be specified via the REFACTR_AUTH_TOKEN environment variable  [string]
      --help             Show help  [boolean]
      --organization-id  Organization this runner belongs to  [string] [required]

Missing required argument: organization-id
"
`;

exports[`refactrctl remove throws on missing subcommand 1`] = `
"refactrctl remove

Commands:
  refactrctl remove pipeline <pipeline-id>      Delete specified pipeline
  refactrctl remove job <job-id>                Delete specified job
  refactrctl remove project <project-id>        Delete specified project
  refactrctl remove runner <runner-id>          Delete specified runner
  refactrctl remove credential <credential-id>  Delete specified runner

Options:
      --version     Show version number  [boolean]
  -v, --verbose     Print detailed output  [boolean]
      --format      Specifies output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --address     Address of the Refactr server. This can also be specified via the REFACTR_ADDRESS environment variable  [string] [required] [default: \\"https://api.refactr.it/v1\\"]
      --auth-token  Authentication token. This can also be specified via the REFACTR_AUTH_TOKEN environment variable  [string]
      --help        Show help  [boolean]

Command must be specified.
"
`;
